core/src/main/java/org/openapi4j/schema/validator/v3/DiscriminatorValidator.java

```diff
@@ -151,6 +161,7 @@ private void setupAllOfDiscriminatorSchemas(final ValidationContext<OAI3> contex
                 discriminatorNode = reference.getContent().get(DISCRIMINATOR);
                 if (discriminatorNode != null) {
                     setupAllOfDiscriminatorSchemas(schemaNode, refNode, reference, schemaParentNode, parentSchema);
+                    addDiscriminatorMappingReferences(discriminatorNode, reference);
```

Comment:
does something like this also need to be done for the setup AnyOf/OneOf case?

Reply:
The adding of these references actually broke something for other validations, so removed the code here. And did a check if the reference actually exists of it cannot be found in the 'original' reference registry